Doc/library/tarfile.rst

```diff
@@ -838,31 +840,37 @@ A ``TarInfo`` object has the following public data attributes:
       attribute.
 
 .. attribute:: TarInfo.chksum
+   :type: int
 
    Header checksum.
 
 
 .. attribute:: TarInfo.devmajor
+   :type: int
 
    Device major number.
 
 
 .. attribute:: TarInfo.devminor
+   :type: int
 
    Device minor number.
 
 
 .. attribute:: TarInfo.offset
+   :type: int
 
    The tar header starts here.
 
 
 .. attribute:: TarInfo.offset_data
+   :type: int
 
    The file's data starts here.
 
 
 .. attribute:: TarInfo.sparse
+   :type: Optional[list]
```

Comment:
Ran into trouble trying to test this, so not confident on this type. I see that it's listed as `bytes` [here](https://github.com/python/typeshed/blob/3195d865382122806ebd1da62d7844c50e90b3af/stdlib/tarfile.pyi#L386), but there's also an `append`:

 https://github.com/python/cpython/blob/82cd8fee31823b560e664f81b430a9186d6019dd/Lib/tarfile.py#L1389-L1392 

Alternatively, it can be left out this time around and investigated in a separate ticket.

```suggestion
```

Reply:
Yeah, the type is not very useful here. Not until we have docs about what's *inside* the list.

But I don't think adding those docs is necessary. With sparse files, IMO the focus should be on Python reading/writing them correctly, rather than exposing details on how they're stored.

Reply:
I feel that if the attribute is documented, then it'd also make sense to keep the type since it may be used. Or did you mean that the entire attribute should not be documented?

Reply:
IMO, the way it's documented now, the only way you can use it is to copy it unchanged to another `TarInfo` (and hope that it's compatible).
The type doesn't help you use it.

But, let's leave it to a separate ticket. Even without the addition, the PR is an improvement.